### DISCUSSION TITLE
Welcome to EnvKey!

### DISCUSSION THREAD

**Discussion by danenania** [envkey/envkey#2] [Announcements]
👋 Hey there,

Thanks for stopping by. I’m Dane, EnvKey’s founder. 

Don't hesitate for a nanosecond to jump in and ask a question, post some feedback, or share your 42 cents on what would make EnvKey better. 

All are welcome!

**Comment by cxostevebenfield** [comment]
Congratulations on v2!! We love envkey and you just made it tons better! Love the Blocks, version control/history, and CLI access.

there's a library out there for ansible that we use -- any idea if it will still work? Native support for ansible would be on my list of 'wants'.

Cheers,
Steve Benfield
CTO - CXO Nexus

**Reply by danenania** [reply]
Thanks @cxostevebenfield!

Do you have a link for that ansible library? I remember that someone had created one, but can't seem to find it now with Google. If it wraps envkey-source, I think it should work as long it's using the latest version, though I'll have to take a look at the code to be sure.

**Reply by cxonexus-github-admin** [reply]
https://github.com/hactar-is/ansible-envkey 

Basically we need to assign environment variable values to ansible variables without having to call envkey before ansible-playbook is called.

But it does appear that we won't need this anymore because of the new envkey CLI -- we should be able to use it to get what we need as long as the output from the command isn't always formatted.

**Comment by drcongo** [comment]
Hey @cxonexus-github-admin - we wrote that Ansible library, and we still use EnvKey so at some point in the coming weeks we will indeed be ensuring it still works on V2. Although the CLI might remove some of the need for it, I also don't fancy rewriting all our Ansible roles! We're also actually in the process of switching to [PyInfra](https://pyinfra.com) from Ansible but that very much depends on me getting the time to spend on it so could be a while away.

And @danenania, thank you so much for listening to your customers on the v2 pricing. Just read the email and I think you've nailed it this time. Actually looking forward to migrating us to v2 now!

**Reply by danenania** [reply]
I'm glad to hear that you're happier with the new pricing @drcongo! And please let me know if I can help at all with getting your Ansible library working with v2 (I'd guess it won't need major changes) or with integrating with pyinfra.

**Comment by mabushey** [comment]
While the support for Envkey has been awesome, it's been very problematic and quirky (v1, no idea about v2). We upgraded to Mozilla Sops and it's working much, much better for us. Sops has a superior workflow and it's also Soc 2 compliant. :)

**Reply by danenania** [reply]
Thanks for the feedback @mabushey. I'd love to hear more details and, if you're game to try it, find out whether v2 might potentially address some of the areas where you found v1 lacking. 

Sops is good, but it does have some drawbacks imho--manual key management, manual encryption/decryption, storing encrypted data in the repo, lack of UI, lack of de-duplication features, etc. Still, there are many worse choices out there, so I'm glad that it's working for you!

EnvKey is also SOC2 compliant now if you self-host it, given that it runs on AWS with no external dependencies. I'd say it puts you in a better posture for a SOC2 than a tool like Sops since the security of Sops depends on getting key management right, which isn't easy.

**Comment by drcongo** [comment]
Hey @danenania, finally going through the docs on migrating to v2 and found a load of broken links here: https://docs-v2.envkey.com/docs/migrating-from-v1#language-libraries-for-node-python-go-and-rubyrails - these all 404 for me...

[envkey-node v2](https://github.com/envkey/envkey/public/sdks/languages-and-frameworks/envkey-node)
[envkey-python v2](https://github.com/envkey/envkey/public/sdks/languages-and-frameworks/envkey-python)
[envkey-ruby v2](https://github.com/envkey/envkey/public/sdks/languages-and-frameworks/envkey-ruby)
[envkeygo v2](https://github.com/envkey/public/sdks/languages-and-frameworks/envkeygo)
[envkey-webpack v2](https://github.com/envkey/public/sdks/languages-and-frameworks/webpack)

**Reply by danenania** [reply]
Thanks @drcongo, those links are now fixed.
